src/shared/sleep-config.c

```diff
@@ -134,6 +134,54 @@ int read_battery_capacity(void) {
         return battery_capacity;
 }
 
+int get_discharge_rate(void) {
+        _cleanup_free_ const char *dirpath;
```

Comment:
this can't be `const`, same in the other function, and also they need to be initialized to NULL